pkg/client/client.go

```diff
@@ -193,6 +193,27 @@ func New(ctx context.Context, baseUrl string, bitbucketClient *DataCenterClient)
 	return &dc, nil
 }
 
+func GetCustomErr(req *http.Request, resp *http.Response, err error) *BitbucketError {
+	bbErr := &BitbucketError{
+		ErrorMessage:     err.Error(),
+		ErrorDescription: err.Error(),
+		ErrorLink:        req.URL.String(),
+	}
+
+	if resp != nil {
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			bbErr.ErrorSummary = fmt.Sprintf("Error reading response body %s", err.Error())
+			return bbErr
```

Comment:
bbErr.ErrorCode should be set before we return. I recommend setting it before line 204.

Reply:
Fixed